pkg/descheduler/utils/sorter/scorer.go

```diff
@@ -50,6 +67,19 @@ func mostRequestedScore(requested, capacity int64) int64 {
 	return (requested * 1000) / capacity
 }
 
+// mostRequestedScorePod The closer the request is to the capacity, the higher the score.
+func mostRequestedScorePod(requested, capacity int64) float64 {
+	if capacity == 0 {
+		return 0
+	}
+	ratio := float64(requested) / float64(capacity)
+	if ratio >= 1 {
+		return 1 / ratio
+	} else {
+		return -1 / ratio
```

Comment:
What is the meaning of **negative** here? I think if a pod's `requested` resource is lower than `capacity`, we can just give it a small score but not a negative one.

Reply:
> What is the meaning of **negative** here? I think if a pod's `requested` resource is lower than `capacity`, we can just give it a small score but not a negative one.

Negative numbers mean that the score output when ratio is greater than or equal to 1 must be greater than the score output when ratio is less than 1. I tried to think about how to give a positive score when requested is less than capacity, and the score is smaller than the score when requested is greater than or equal to capacity, but unfortunately, no conclusion was reached, so using plural numbers was the fastest and most effective solution I thought at the time.

Reply:
@songtao98 Do you think negative numbers affect code reading or standardization, so do you suggest modifications? Because negative numbers do meet the needs in this scenario, I don't know the main reason why I can suggest modifications.

Reply:
@songtao98 In other words, use positive values ​​for all scores in this scenario. It is difficult to guarantee that the score output when requested>=capacity must be greater than the score output when requested<capacity. If you have a better way to do it, I'd love to hear it.

Reply:
> @songtao98 Do you think that negative numbers affect code reading or do not comply with specifications, so do you suggest modifications? Because negative numbers do meet the functional requirements in this scenario, I don't know the main reason for your suggested modification.

I get this. There is only one possible but extreme bad case: Because when we calculate the score of one pod, we calculate through all resources. That means, if a pod get a lower score on one resource(CPU), it could get a higher **final** score than other pods because it get higher scores on **all other** resources(Memory, etc.).  If you make it negative, I think this pod will totally lost a chance to win back. This is why I suggest we find a positive but small score when ratio is < 1 **if possible**. 

An ugly solution is to use combined `piecewise functions. ` For example, 

```
if ratio >= 1 {
		return 1 / ratio + 1 // score is in (1, +∞)
	} else {
		return  ratio // score is in [0, 1)
```

Anyway, I'm doubting this, too. And I'm sure to be open to hear your opinion about whether this is over-designed. :)

Reply:
@songtao98 After seeing your description, I understand the reason. Indeed, your consideration is more comprehensive and reasonable. I will resubmit using the piecewise functions. Thank you for your suggestion. I have benefited a lot.

Reply:
@songtao98 
Submitted, please re-evaluate

Reply:
@songtao98 Is there anything else that needs to be corrected?

Reply:
> @songtao98 Is there anything else that needs to be corrected?

Not for now, this PR looks good to me. But as I said in another comment, there might be a problem with another PR. We will work as soon as possible to finish review and merge it. After that, you might need to modify the way you Sub the resource from the threshold.

Reply:
> > @songtao98 Is there anything else that needs to be corrected?
> 
> Not for now, this PR looks good to me. But as I said in another comment, there might be a problem with another PR. We will work as soon as possible to finish review and merge it. After that, you might need to modify the way you Sub the resource from the threshold.

It means that after merging the PR of https://github.com/koordinator-sh/koordinator/pull/2066, this PR needs to be adapted before it can be passed, right?

Reply:
> > > @songtao98 Is there anything else that needs to be corrected?
> > 
> > 
> > Not for now, this PR looks good to me. But as I said in another comment, there might be a problem with another PR. We will work as soon as possible to finish review and merge it. After that, you might need to modify the way you Sub the resource from the threshold.
> 
> It means that after merging the PR of #2066, this PR needs to be adapted before it can be passed, right?

Yes it is. I'll help with your adaption anytime you need. And I'll try to push this process to be quick.

Reply:
> > > > @songtao98 Is there anything else that needs to be corrected?
> > > 
> > > 
> > > Not for now, this PR looks good to me. But as I said in another comment, there might be a problem with another PR. We will work as soon as possible to finish review and merge it. After that, you might need to modify the way you Sub the resource from the threshold.
> > 
> > 
> > It means that after merging the PR of #2066, this PR needs to be adapted before it can be passed, right?
> 
> Yes it is. I'll help with your adaption anytime you need. And I'll try to push this process to be quick.

I will listen to your arrangements and thank you for your guidance.

Reply:
@songtao98 The latest main branch was merged, and the E2E test failed.

Reply:
> @songtao98 The latest main branch was merged, and the E2E test failed.

will be fixed by #2101

Reply:
> > @songtao98 The latest main branch was merged, and the E2E test failed.
> 
> will be fixed by #2101

”leader election lost“，
I'm afraid you need to re-execute E2E here.

Reply:
> > > @songtao98 The latest main branch was merged, and the E2E test failed.
> > 
> > 
> > will be fixed by #2101
> 
> ”leader election lost“， I'm afraid you need to re-execute E2E here.

Sorry for the late reply. #2066 is now merged into main. You can now rebase main and modify your PR to adapt to the prodThreshold. After that your push should auto execute E2E here.

Reply:
> > > > @songtao98 The latest main branch was merged, and the E2E test failed.
> > > 
> > > 
> > > will be fixed by #2101
> > 
> > 
> > ”leader election lost“， I'm afraid you need to re-execute E2E here.
> 
> Sorry for the late reply. #2066 is now merged into main. You can now rebase main and modify your PR to adapt to the prodThreshold. After that your push should auto execute E2E here.

What exactly should I do? I don’t know the implementation of this PR: https://github.com/koordinator-sh/koordinator/pull/2066